tests/test_api.py

```diff
@@ -730,3 +732,50 @@ def test_delete_build_auth(testclient):
 
     # r = schema.APIResponse.parse_obj(response.json())
     # assert r.status == schema.APIStatus.ERROR
+
+
+def test_api_cancel_build(testclient):
+    build_id = 1
+
+    testclient.login()
+    response = testclient.put(f"api/v1/build/{build_id}")
+    r = schema.APIPostSpecification.parse_obj(response.json())
+    assert r.status == schema.APIStatus.OK
+
+    new_build_id = r.data.build_id
+
+    # delay to ensure the build kicks off
+    build_timeout = 180
+    building = False
+    start = time.time()
+    while time.time() - start < build_timeout:
+        response = testclient.get(f"api/v1/build/{new_build_id}")
+        response.raise_for_status()
+
+        r = schema.APIGetBuild.parse_obj(response.json())
+        if r.data.status == schema.BuildStatus.BUILDING.value:
+            building = True
+            break
+        time.sleep(30)
+
+    assert building is True
+
+    # The new build should have kicked off, so now we will request to cancel it
+    response = testclient.put(f"api/v1/build/{new_build_id}/cancel")
+
+    r = schema.APIResponse.parse_obj(response.json())
+
+    assert r.status == schema.APIStatus.OK
+    assert "canceled" in r.message
```

Comment:
Formatting: remove newlines here, would be easier to read near `r`.
Check the full error message in the assert.